File: overlappingIntervals.ts
function eraseOverlapIntervals(intervals: number[][]): number {
    if (intervals.length <= 0){
        return 0;
    }
    intervals.sort((a,b) => a[1]-b[1]);

    let count = 0;
    let prev = intervals[0][1];

    for(let i = 1; i < intervals.length; i++){
        if (intervals[i][0] < prev){
            count++;
        }
        else{
            prev = intervals[i][1]
        }
    }
    return count;
};

File: overLappingIntervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda x: x[1])
        n = len(intervals)
        count = 0
  
        currR = intervals[0][1]
        for i in range(1,n):
            if intervals[i][0] < currR:
                count += 1
            else:
                currR = intervals[i][1]
       
        return count

        
